### ISSUE TITLE
castNonNull required even after if != null check

### ISSUE THREAD

**Issue by jpschewe** [typetools/checker-framework#4566]
Using checker 3.12.0.
I would have expected checker to recognize that in this particular case that `mDescription` cannot be null and therefore not give a warning on calling `createTextNode`.

```java
import org.checkerframework.checker.nullness.qual.Nullable;

import org.w3c.dom.Document;
import org.w3c.dom.Element;

public class IfNotNull {

  /**
   * The XML attribute to write the description to.
   */
  public static final String DESCRIPTION_TAG_NAME = "description";
  
  private @Nullable String mDescription;

  public Element toXml(final Document doc) {
    final Element ele = doc.createElement("tag");
    
    if (null != mDescription) {
      final Element descriptionEle = doc.createElement(DESCRIPTION_TAG_NAME);
      descriptionEle.appendChild(doc.createTextNode(mDescription));
      ele.appendChild(descriptionEle);
    }
    
    return ele;
  }
 
}
```

**Comment by jpschewe** [comment]
This one appears to be much the same as #4571. The fact that a method is executed after the if check and before the use of `mDescription` is causing the warning. If I change my code like below the warning goes away.

```java
package net.mtu.eggplant.checker;

import org.checkerframework.checker.nullness.qual.Nullable;

import org.w3c.dom.Document;
import org.w3c.dom.Element;

public class IfNotNull {

  /**
   * The XML attribute to write the description to.
   */
  public static final String DESCRIPTION_TAG_NAME = "description";
  
  private @Nullable String mDescription;

  public Element toXml(final Document doc) {
    final Element ele = doc.createElement("tag");
    
    if (null != mDescription) {
      final String description = mDescription;
      final Element descriptionEle = doc.createElement(DESCRIPTION_TAG_NAME);
      descriptionEle.appendChild(doc.createTextNode(description));
      ele.appendChild(descriptionEle);
    }
    
    return ele;
  }
  

}
```
